### ISSUE TITLE
Threat HUD tinting TaN's HUD

### ISSUE THREAD

**Issue by Snownee** [Laike-Endaril/Dynamic-Stealth#4]
![image](https://user-images.githubusercontent.com/16059084/50479534-0325aa80-0a12-11e9-8722-abcfb485a7e5.png)

**Comment by Laike-Endaril** [comment]
Admittedly I have not yet gotten around to trying Tough as Nails.  Is the circle in the middle the wrong color (ie. "bleeding" from my HUD's red text)?

**Comment by Snownee** [comment]
Yes it should be light blue

**Comment by Laike-Endaril** [comment]
My HUD is very simple and only uses the vanilla text drawing function drawString(), which allows a color as an argument.  This being the case, I would say that the function is setting the openGL draw color and not resetting it back to whatever it was before, which may sound bad, but is actually fairly common.

Basically, it's almost as if TaN is setting their color too early (or possibly not at all), allowing my code to run between when their color is set and when their color is used.  I might be able to "fix" it from my end by saving the current draw color, drawing my HUD, and then setting the draw color back to what it was, but that will *only fix the issue in relation to my mod*, which is to say that any *other mod* could still cause the issue, if it happens to draw a GUI/HUD in the same way that I did.

TL;DR this is probably technically a TaN issue, but me being the way I am, I will look into it anyway :P

**Comment by Laike-Endaril** [comment]
https://github.com/Glitchfiend/ToughAsNails/issues/607

I will leave this open for now (mostly to prevent re-posts of the issue).  If anyone notices that the issue is fixed in a future version of TaN, please notify me with a post in this thread!

**Comment by Laike-Endaril** [comment]
I've fixed this issue, at least in relation to my own mod, in version 011 of my mod:
[https://minecraft.curseforge.com/projects/dynamic-stealth/files](https://minecraft.curseforge.com/projects/dynamic-stealth/files)

I am *NOT* closing the issue I opened on *their* end, as the root cause is still that they are not setting their color before drawing their GUI / HUD element, and this same thing could easily happen between TaN and some other random mod

But hey, it works with DS now! ;)

**Comment by WenXin20** [comment]
I noticed the same thing happens when holding the guide book from Cyclic Magic, which uses the mod Guide Api for the book.

**Comment by Laike-Endaril** [comment]
@WenXin20 I don't see a mod named cylic magic on curseforge, though I do see one just named "cyclic". 
 Is that the mod you're referring to?
If so, which versions of DS, Cyclic, and Guide API do you have (so I can run a test)?

**Comment by WenXin20** [comment]
Yes, that's the one. I think the mod id is cyclicmagic. I'm using DS v1.12.2.051, Cyclic v1.17.16, and Guide API v2.1.8-63.

**Comment by Laike-Endaril** [comment]
Thanks, I'll check this out after I do a couple other things

**Comment by WenXin20** [comment]
Ok, thanks.-----------------------Sincerely, David NewvilleOn Feb 18, 2019 5:42 PM, Laike-Endaril <notifications@github.com> wrote:Thanks, I'll check this out after I do a couple other things

**Comment by Laike-Endaril** [comment]
Running only the 3 mods mentioned, I'm unable to reproduce the issue so far (same versions).  Any additional details would be helpful (eg. screenshot).

Also, if you could test with only the 3 mods on your end and see if it happens or not.  If not, another mod is contributing to the issue as well.

**Comment by WenXin20** [comment]
Here's an image from yesterday, but I probably won't be able to get back on the game for a couple more days:
![2019-02-18_10 54 41](https://user-images.githubusercontent.com/12000593/52988358-889b5a80-33c4-11e9-8e4d-093db59c791c.png)

**Comment by Laike-Endaril** [comment]
I'm actually not sure what's wrong in that image tbh

Edit: Which part of that screenshot is the wrong color?

**Comment by WenXin20** [comment]
The item I am holding. The majority of the item is purple, but there are parts that should be orange/yellow/white, like the inventory item. It looks like the whole item somehow has a red tinge to it now.

**Comment by Laike-Endaril** [comment]
Ah yes, I see what you mean now.

This was due to a feature I recently added which caused several issues and ended up having to be removed for now.  Ie. this is already fixed in the newest version (054).

I made sure by doing a test with version 051 (and was able to reproduce), and then testing with 054 (book fixed).  Grab the latest version and you should be good!

**Comment by WenXin20** [comment]
Ok, thanks!-----------------------Sincerely, David NewvilleOn Feb 19, 2019 3:32 PM, Laike-Endaril <notifications@github.com> wrote:Ah yes, I see what you mean now.
This was due to a feature I recently added which caused several issues and ended up having to be removed for now.  Ie. this is already fixed in the newest version (054).
I made sure by doing a test with version 051 (and was able to reproduce), and then testing with 054 (book fixed).  Grab the latest version and you should be good!
